service/controller/resource/namespaceauth/create.go

```diff
@@ -83,6 +83,16 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			return microerror.Mask(err)
 		} else {
 			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("role binding %#q already exists", newRoleBinding.Name))
+
+			if role.targetGroup != existingRoleBinding.Subjects[0].Name {
+				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating role binding %#q", newRoleBinding.Name))
+				_, err := r.k8sClient.RbacV1().RoleBindings(namespace.Name).Update(newRoleBinding)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("role binding %#q has been updated", newRoleBinding.Name))
+
+			}
```

Comment:
I'm not blocking but I'd like to have a nicer flow here to make future changes nice.

Suggestion:

```go
} else if needsUpdate(role, existingRoleBinding) {
	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating role binding %#q", newRoleBinding.Name))
	_, err := r.k8sClient.RbacV1().RoleBindings(namespace.Name).Update(newRoleBinding)
	if err != nil {
		return microerror.Mask(err)
	}
	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("role binding %#q has been updated", newRoleBinding.Name))

} else {
	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("role binding %#q already exists", newRoleBinding.Name))
}

func needsUpdate(role, existingRoleBinding) bool {
	role.targetGroup != existingRoleBinding.Subjects[0].Name
}
```

Then in the future if we decide update conditions change we need to update only `needsUpdate` func w/o changing this big if/else.

Reply:
smart!